crates/core/src/db/query.rs

```diff
@@ -0,0 +1,297 @@
+use crate::db::datastore::locking_tx_datastore::committed_state::CommittedState;
+use crate::db::datastore::locking_tx_datastore::tx::TxId;
+use crate::db::datastore::locking_tx_datastore::MutTxId;
+use crate::db::datastore::traits::Tx;
+use spacetimedb_data_structures::map::IntMap;
+use spacetimedb_execution::iter::{Datastore, DeltaScanIter};
+use spacetimedb_primitives::{ColList, IndexId, TableId};
+use spacetimedb_sats::{AlgebraicValue, ProductValue};
+use spacetimedb_table::blob_store::BlobStore;
+use spacetimedb_table::btree_index::BTreeIndex;
+use spacetimedb_table::table::{IndexScanIter, RowRef, Table, TableScanIter};
+use std::ops::RangeBounds;
+
+pub trait DatastoreEx {
+    fn get_committed_state(&self) -> &CommittedState;
+    fn get_tables(&self) -> &IntMap<TableId, Table>;
+}
+
+impl Datastore for TxId {
+    fn delta_scan_iter(&self, _table_id: TableId) -> DeltaScanIter {
+        DeltaScanIter::empty_iter()
+    }
+
+    fn table_scan_iter(&self, table_id: TableId) -> TableScanIter {
```

Comment:
All this methods requiere `unwrap` and assume the acquiring of the `iter` will not fail. I think we should make them return a `Result` so we can report `Index/Table not found`